From b5597a9ac4fbb910013942a722b6bff70088681d Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Thu, 14 Aug 2014 15:05:21 -0400
Subject: [firewire] ohci: beautify some macro definitions

Message-id: <1408028722-23447-2-git-send-email-nhorman@redhat.com>
Patchwork-id: 87733
O-Subject: [RHEL7.1 PATCH 1/2] firewire: ohci: beautify some macro definitions
Bugzilla: 994878
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

a) Sort device IDs by vendor -- device -- revision.

b) Write quirk flags in hexadecimal.  This affects the user-visible
output of "modinfo firewire-ohci".  Since more flags have been added
recently, it is now easier to cope with them in hexadecimal represen-
tation.  Besides, the device-specific combination of quirk flags is
shown in hexadecimal in the kernel log too.  (And firewire-sbp2
presents its own quirk flags in modinfo as hexadecimals as well.)

Signed-off-by: Stefan Richter <stefanr@s5r6.in-berlin.de>
(cherry picked from commit 0dbe15f88be5b2cdf4ca4145797861dfb0d583a5)

BZ: 1096023
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7839135
Testing: I don't have hardware to test this with, but the reporter (Jan Stancek)
does, and confirmed that it solved the reported issues

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/firewire/ohci.c b/drivers/firewire/ohci.c
index afb701e..66a63b4 100644
--- a/drivers/firewire/ohci.c
+++ b/drivers/firewire/ohci.c
@@ -271,6 +271,7 @@ static inline struct fw_ohci *fw_ohci(struct fw_card *card)
 
 static char ohci_driver_name[] = KBUILD_MODNAME;
 
+#define PCI_VENDOR_ID_PINNACLE_SYSTEMS 0x11bd
 #define PCI_DEVICE_ID_AGERE_FW643 0x5901
 #define PCI_DEVICE_ID_CREATIVE_SB1394 0x4001
 #define PCI_DEVICE_ID_JMICRON_JMB38X_FW 0x2380
@@ -278,17 +279,16 @@ static char ohci_driver_name[] = KBUILD_MODNAME;
 #define PCI_DEVICE_ID_TI_TSB12LV26 0x8020
 #define PCI_DEVICE_ID_TI_TSB82AA2 0x8025
 #define PCI_DEVICE_ID_VIA_VT630X 0x3044
-#define PCI_VENDOR_ID_PINNACLE_SYSTEMS 0x11bd
 #define PCI_REV_ID_VIA_VT6306  0x46
 
-#define QUIRK_CYCLE_TIMER  1
-#define QUIRK_RESET_PACKET  2
-#define QUIRK_BE_HEADERS  4
-#define QUIRK_NO_1394A   8
-#define QUIRK_NO_MSI   16
-#define QUIRK_TI_SLLZ059  32
-#define QUIRK_IR_WAKE   64
-#define QUIRK_PHY_LCTRL_TIMEOUT  128
+#define QUIRK_CYCLE_TIMER  0x1
+#define QUIRK_RESET_PACKET  0x2
+#define QUIRK_BE_HEADERS  0x4
+#define QUIRK_NO_1394A   0x8
+#define QUIRK_NO_MSI   0x10
+#define QUIRK_TI_SLLZ059  0x20
+#define QUIRK_IR_WAKE   0x40
+#define QUIRK_PHY_LCTRL_TIMEOUT  0x80
 
 /* In case of multiple matches in ohci_quirks[], only the first one is used. */
 static const struct {
-- 
1.7.1